src/app/applications/guards/applications-access.guard.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable, inject } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { UserService } from '@services/user.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApplicationsAccessGuard implements CanActivate {
+  private readonly userService = inject(UserService);
+  private readonly router = inject(Router);
+
+  canActivate(): boolean {
+    const permissions = this.userService.user?.permissions ?? [];
+    const hasPermission = permissions.includes('Applications:ListApplications');
+    
+    if (!hasPermission) {
+      this.router.navigate(['/dashboard']);
+      return false;
+    }
+    
+    return true;
+  }

```

Comment:
Could you create an abstract class with just an abstract member `permission`, which would represent in this case "Applications:ListApplications".

We would have in the end:
```ts
export class ApplicationsAccessGuard extends AbstractAcessGuard {
  permission = 'Applications:ListApplications';
}